workflows/trust4/trust4.wdl

```diff
@@ -0,0 +1,298 @@
+        from fnmatch import fnmatch
+        from subprocess import check_call, CalledProcessError, DEVNULL, STDOUT
+        import sys
+
+        Path = os.path.join
+
+        if '~{input_fastqs_directories}' and '~{input_bam}':
+            sys.exit("Please provide either Fastq OR BAM as input (not both)")
+
+        if '~{input_fastqs_directories}':
+            if '~{pe_read1_fastq_pattern}' and '~{pe_read2_fastq_pattern}':
+                se_fastq_pattern = ''
+            else:         
+                se_fastq_pattern = '~{se_fastq_pattern}'
+
+        if '~{input_fastqs_directories}':
+            fastq_dirs = []
+            for i, directory in enumerate('~{input_fastqs_directories}'.split(',')):
+                directory = re.sub('/+$', '', directory) # remove trailing slashes
+                target = "~{sample_id}_" + str(i)
+                try:
+                    call_args = ['strato', 'exists', '--backend', '~{backend}', directory + '/~{sample_id}/']
+                    print(' '.join(call_args))
+                    check_call(call_args, stdout=DEVNULL, stderr=STDOUT)
+                    call_args = ['strato', 'sync', '--backend', '~{backend}', '-m', directory + '/~{sample_id}', target]
+                    print(' '.join(call_args))
+                    check_call(call_args)     
+                except CalledProcessError:
+                    if not os.path.exists(target):
+                        os.mkdir(target)
+                    call_args = ['strato', 'cp', '--backend', '~{backend}', '-m', directory + '/~{sample_id}*' , target + '/']
+                    print(' '.join(call_args))
+                    check_call(call_args)
+
+                fastq_dirs.append(target)
+
+        if '~{input_bam}':
+            target = '~{sample_id}_' + str(i)
+            if not os.path.exists(target):
+                os.mkdir(target)
+            call_args = ['strato', 'cp', '--backend', '~{backend}', '-m', ~{input_bam} , target + '/']
+            print(' '.join(call_args))
+            check_call(call_args)
+
+
+        call_args = ['run-trust4', '-f', 'genome_dir/bcrtcr.fa' ,
+                     '--ref', 'genome_dir/IMGT+C.fa', '-t', '~{num_cpu}',
+                     '--od', 'trust4_~{sample_id}', '-o', '~{sample_id}']
+        
+        if '~{barcode_whitelist}':
+            call_args.extend(['--barcodeWhitelist', '~{barcode_whitelist}'])
+        if '~{skipMateExtension}':
+            call_args.append('--skipMateExtension')
+        if '~{mateIdSuffixLen}':
+            call_args.extend(['--mateIdSuffixLen', '~{mateIdSuffixLen}'])
+        if '~{noExtraction}':
+            call_args.append('--noExtraction')
+        if '~{repseq}':
+            call_args.extend(['--repseq', '~{repseq}'])
+        if '~{outputReadAssignment}':
+            call_args.append('--outputReadAssignment')
+
+        if '~{barcode_range}':
+            call_args.extend(['--barcode_range', ' '.join(~{barcode_range}'.split(','))])
```

Comment:
To be safe, please make each parameter as a separate item in the list.

For example `--barcodeRange 0 15 +` should be ['--barcodeRange', '0', '15', '+'].